### DISCUSSION TITLE
When I build with CGO_ENABLED=0, I still get a dynamic executable

### DISCUSSION THREAD

**Discussion by alexflint** [ebitengine/purego#292] [General]
Thanks for this amazing library.

When I compile a program that uses purego, I get a dynamic executable:

```bash
$ CGO_ENABLED=0 go build -o bin
$ ldd bin
	linux-vdso.so.1 (0x00007fff66997000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007a113a0d1000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007a1139e00000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007a113a0cc000)
	/lib64/ld-linux-x86-64.so.2 (0x00007a113a0f3000)
```

When I comment purego out of my program, I get a static executable.
```bash
$ CGO_ENABLED=0 go build -o bin
$ ldd bin                                                                       
	not a dynamic executable
```

Is this expected? If so, how actually does this happen? I thought a program compiled with CGO_ENABLED=0 would never link to external libraries. Finally, is it possible to use purego while producing a static executable?

**Comment by TotallyGamerJet** [comment]
> Is this expected?

Yes

> If so, how actually does this happen? 

Magical pragmas like [these](https://github.com/ebitengine/purego/blob/907f1d943ac12dc80a36af9f715924cb95cc3be2/dlfcn_nocgo_linux.go#L11)

>  I thought a program compiled with CGO_ENABLED=0 would never link to external libraries.

It only means that a C compiler was not used to build the executable allowing cross-compilation to be possible again.

> is it possible to use purego while producing a static executable?

No, purego needs to be able to use libc's `dlopen` and `dlsym` to load symbols into memory.

**Reply by alexflint** [reply]
Fascinating! Thanks for the info and the quick reply!
